crates/sound/src/device.rs

```diff
@@ -564,7 +685,7 @@ impl VhostUserSoundBackend {
             virtio_cfg: VirtioSoundConfig {
                 jacks: 0.into(),
                 streams: 1.into(),
-                chmaps: 0.into(),
+                chmaps: 1.into(),
```

Comment:
Should we start using the length of the vector here (but also for `jacks` and `streams`) instead of wiring the values? (Not for this PR but for future cleanup)

Reply:
Not just that, it seems to me both jack and chmap info (and [pcm params too](https://github.com/virtio-sound/vhost-device/blob/0b4523617fd3ef6b80f31d96ed963fdef4550d06/crates/sound/src/stream.rs#L214)) should eventually come from the audio backend, unless we opt for making this stuff configurable by the user on startup and not allow for probing the host sound card for stuff like that.

Making the virtual sound card independent of the host audio backend means we have to be in charge of resampling in case of incompatible rates/formats, speaker-filling in case of different channels, potentially (bad) channel remapping... On a second thought querying all this stuff from the backend sounds like a better approach.

Reply:
I agree